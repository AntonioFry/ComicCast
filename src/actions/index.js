export const favoriteCharacter = (characterId) => ({
  type: "FAVORITE_CHARACTER",
  characterId
})