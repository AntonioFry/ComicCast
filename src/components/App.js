import React, { Component } from 'react';
import './App.css';
import NavBar from './NavBar/NavBar';
import Home from './Home/Home';
import { Route } from 'react-router-dom';

class App extends Component {
  constructor() {
    super();
    this.state = {
      error: ''
    }
  }

  componentDidMount = async () => {
    try {
      const response = await fetch(`https://www.superheroapi.com/api.php/1983143645120471/621`)
      const result = await response.json();
      console.log(result);
    } catch (error) {
      console.log(error);
    }
  }

  render() {
    return (
      <main>
        <NavBar />
        <Route exact path="/" render={<Home />} />
      </main>
    )
  }

}

export default App;
