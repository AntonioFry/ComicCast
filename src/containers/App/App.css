main {
  background: url('../../images/Background-stock.jpg');
  height: 700px;
}

.loading {
  margin: 100px 40%;
  width: 20%;
}