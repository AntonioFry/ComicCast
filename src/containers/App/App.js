import React, { Component } from 'react';
import './App.css';
import NavBar from '../../components/NavBar/NavBar';
import Home from '../../components/Home/Home';
import { Route } from 'react-router-dom';
import { getCharacters } from '../../api/apiCalls';
import Search from '../../components/Seacrh/Search';
import { connect } from 'react-redux';
import { setCharacters } from '../../actions';

class App extends Component {
  constructor() {
    super();
    this.state = {
      allCharacters: [],
      featuredCharacter: {},
      suggestedCharacters: [],
      error: ''
    }
  }

  getRandomCharacters = async () => {
    try {
      const suggestedCharacters = [];
      for (let i = 0; i < 5; i++) {
        const randomNumber = Math.floor(Math.random() * 26);
        const character = this.state.allCharacters[randomNumber]
        suggestedCharacters.push(character);
      }
      this.setState({ suggestedCharacters });
    } catch (error) {
      this.setState({ error: error.message });
    }
  }

  getFeaturedCharacter = () => {
    const randomNumber = Math.floor(Math.random() * 20);
    const featuredCharacter = this.state.allCharacters[randomNumber];
    this.setState({ featuredCharacter });
  }

  componentDidMount = async () => {
    try {
      const allCharacters = await getCharacters();
      console.log(this.state.allCharacters)
      this.setState({ allCharacters });
      this.getFeaturedCharacter();
      this.getRandomCharacters();
      this.props.setCharacters(allCharacters)
    } catch (error) {
      this.setState({ error: error.message });
    }
  }

  render() {
    return (
      <main>
        <NavBar />
        {!this.props.allCharacters.length && 
        <img className="loading" src={require('../../images/Double-Ring.gif')} alt="a loading icon"/>}
        {this.props.allCharacters.length && <Route exact path="/" render={() => <Home/>} />}
        {/* <Route exact path="/Favorites" render={() => <Favorites />} /> */}
        <Route exact path="/Search" render={() => <Search />} />
      </main>
    )
  }
}

const mapStateToProps = (store) => ({
  allCharacters: store.allCharacters
})

const mapDispatchToProps = (dispatch) => ({
  setCharacters: characters => dispatch(setCharacters(characters))
})

export default connect(mapStateToProps, mapDispatchToProps)(App);
