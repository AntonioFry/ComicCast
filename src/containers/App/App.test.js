import React from 'react';
import { App } from './App';
import { shallow } from 'enzyme';
import { mapStateToProps, mapDispatchToProps } from './App';
import { setCharacters } from '../../actions';

describe('App', () => {

  describe("App component", () => {
    let wrapper;
    let props;
  
    beforeEach(() => {
      props = {
        allCharacters: [{}, {}]
      }
      wrapper = shallow(<App allCharacters={props.allCharacters} />)
    });
  
    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  
    it.skip('should only set state of error if componentDidMount catches error', () => {
      window.fetch = jest.fn().mockImplementation(() => {
        return Promise.resolve({
          ok: false
        })
      });
      wrapper.instance().componentDidMount();
      expect(wrapper.state('error')).toEqual("failed to fetch characters")
    });
  });

  describe('mapStateToProps', () => {

    it('should return an object with all characters', () => {
      const mockStore = {
        allCharacters: [{}, {}, {}, {}, {}],
      }
      const expected = {
        allCharacters: [{}, {}, {}, {}, {}],
      }
      const mappedProps = mapStateToProps(mockStore);
      expect(mappedProps).toEqual(expected);
    });
  });

  describe("mapDispatchToProps", () => {
    it('calls dispatch with setCharacters action', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = setCharacters([{ name: 'Spider-Man' }, { name: 'Thor' }])
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.setCharacters([{ name: 'Spider-Man' }, { name: 'Thor' }])
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });
});