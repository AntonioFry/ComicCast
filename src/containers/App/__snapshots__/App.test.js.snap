// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App component should match the snapshot 1`] = `
<main>
  <NavBar />
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/Search"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/undefined"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/undefined"
    render={[Function]}
  />
</main>
`;
